src/sortStudents.ts

```diff
@@ -1,16 +1,100 @@
-
 export interface Student {
   // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
   // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+function getAverageGrade(grades:number[]): number {
+  return grades.reduce((prev, current) => prev + current, 0) / grades.length;
+}
+
+function sortedByNameAndSurname(
+  objectOfStudents: Student[],
+  order: string,
+) :Student[] {
+  if (order === 'asc') {
+    return objectOfStudents
+      .sort((first, second) => first.surname.localeCompare(second.surname));
+  }
+
+  return objectOfStudents
+    .sort((first, second) => second.surname.localeCompare(first.surname));
+}
+
+function sortedByAverageGrade(
+  objectOfStudents: Student[],
+  order: string,
+): Student[] {
+  if (order === 'asc') {
+    return objectOfStudents
+      .sort((first, second) => getAverageGrade(first.grades)
+        - getAverageGrade(second.grades));
+  }
+
+  return objectOfStudents.sort((first, second) => getAverageGrade(second.grades)
+    - getAverageGrade(first.grades));
+}
+
+function sortedByAge(objectOfStudents : Student[], order: string): Student[] {
+  if (order === 'asc') {
+    return objectOfStudents.sort((first, second) => first.age - second.age);
+  }
+
+  return objectOfStudents.sort((first, second) => second.age - first.age);
+}
+
+function sortedByMarried(
+  objectOfStudents: Student[],
```

Comment:
хіба це `objectOfStudents`? наскільки я знаю, це масив - array. Але краще використовувати просто `students`.
```suggestion
  students: Student[],
```